src/main/steps/common/support-you-need-during-case/summary/handler.ts

```diff
@@ -0,0 +1,144 @@
+import { CaseWithId } from '../../../../app/case/case';
+import { ReasonableAdjustments } from '../../../../app/case/definition';
+import { LANGUAGE_INTERPRETER, NO_HEARINGS } from '../../../../steps/constants';
+import {
+  CA_DA_COMMUNICATION_HELP,
+  CA_DA_COURT_HEARING_COMFORT,
+  CA_DA_COURT_HEARING_SUPPORT,
+  CA_DA_DOCUMENTS_SUPPORT,
+  CA_DA_TRAVELLING_TO_COURT,
+} from '../../../../steps/urls';
+
+export function filterSelectedUrls(userCase: Partial<CaseWithId>, urls: object): void {
+  reasonableAdjustment(userCase, urls);
+
+  if (!userCase?.attendingToCourt?.includes(NO_HEARINGS)) {
+    userCase.hearingDetails = '';
+  }
+
+  if (!userCase?.languageRequirements?.includes(LANGUAGE_INTERPRETER)) {
+    userCase.languageDetails = '';
+  }
+  if (!userCase?.safetyArrangements?.includes('other')) {
+    userCase.safetyArrangementsDetails = '';
+  }
+  //resonable adjustment
+  dataCleanupReasonableadjustment(userCase);
+}
+function dataCleanupReasonableadjustment(userCase: Partial<CaseWithId>) {
+  if (!userCase?.reasonableAdjustments?.includes('docsformat')) {
+    delete userCase.docsSupport;
+  }
+  if (!userCase?.reasonableAdjustments?.includes('commhelp')) {
+    delete userCase.helpCommunication;
+  }
+  if (!userCase?.reasonableAdjustments?.includes('hearingsupport')) {
+    delete userCase.courtHearing;
+  }
+  if (!userCase?.reasonableAdjustments?.includes('hearingcomfort')) {
+    delete userCase.courtComfort;
+  }
+  if (!userCase?.reasonableAdjustments?.includes('travellinghelp')) {
+    delete userCase.travellingToCourt;
+  }
+  //docSupport
+  dataCleanupDocSupport(userCase);
+  //comunication help
+  dataCleanupCommunicationHelp(userCase);
+  //Hearing Support
+  dataCleanupHearingSupport(userCase);
+  //court comfort
+  dataCleanupCourtComfort(userCase);
+  //travel help
+  dataCleanupTravelHelp(userCase);
+}
+function dataCleanupCourtComfort(userCase: Partial<CaseWithId>) {
+  if (!userCase?.courtComfort?.includes('appropriatelighting')) {
+    userCase.lightingProvideDetails = '';
+  }
+  if (!userCase?.courtComfort?.includes('other')) {
+    userCase.otherProvideDetails = '';
+  }
+}
+
+function dataCleanupTravelHelp(userCase: Partial<CaseWithId>) {
```

Comment:
Prefix process for all dataCleanup and add Data at the end for all

for eg: change dataCleanupCourtComfort to processCourtComfortData.